src/sparsebase/preprocess/preprocess.h

```diff
@@ -938,109 +962,186 @@ class MetisPartition : public PartitionPreprocessType<IDType> {
     int64_t ufactor = 30;
   };
 };
+
+template <typename IDType, typename NNZType, typename ValueType>
+class MetisReorder : public ReorderPreprocessType<IDType>{
+public:
+  struct MetisReorderParams : PreprocessParams {
+    int64_t ctype = metis::METIS_CTYPE_RM;
+    int64_t rtype = metis::METIS_RTYPE_SEP2SIDED;
+    int64_t nseps = 1;
+    int64_t numbering = 0;
+    int64_t niter = 10;
+    int64_t seed = 42;
+    int64_t no2hop = 0;
+    int64_t compress = 0;
+    int64_t ccorder = 0;
+    int64_t pfactor = 0;
+    int64_t ufactor = 30;
+  };
+  typedef MetisReorderParams ParamsType;
+  MetisReorder();
```

Comment:
Can you add a constructor that takes a params object? It's needed for the base function. Also, can you add a test that uses ReoredrBase with METIS reorder (similar to the `ReorderBase.RCMReorder` test in `preprocess_tests.cc`.